### PR TITLE
crimson/test: add CBT based perf tests

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
i am still playing around with these scripts. but want to have some comments from early readers.

File: src/script/run-cbt.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+example:
+  $prog_name -a /tmp radosbench-write.yaml
+EOF
+}
+
+MDS=0 MGR=1 OSD=3 MON=1 ../src/vstart.sh -n --without-dashboard --memstore -X \
+   -o "memstore_device_bytes=34359738368" \
+   --crimson --nodaemon --redirect-output \
+   --osd-args "--memory 4G --smp 1 --cpuset 0"
```
Comment (rzarzynski) [inline src/script/run-cbt.sh]: In the matter of `--cpuset 0`: aren't we overriding the behavior introduced to `vstart,sh` by the first commit?

File: src/test/crimson/cbt/t2c.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        benchmarks = config.get('benchmarks', [])
+        monitoring_profiles = config.get('monitoring_profiles', {})
+        return dict(cluster=self._create_cluster_config(cluster),
+                    benchmarks=benchmarks,
+                    monitoring_profiles=monitoring_profiles)
+
+    def _create_cluster_config(self, cluster):
+        # prepare the "cluster" section consumed by CBT
+        localhost = 'gen8'
+        num_osds = cluster.get('osds_per_node', 1)
```
Comment (cyx1231st) [inline src/test/crimson/cbt/t2c.py]: is it default to 3 osds per node?

File: src/test/crimson/cbt/t2c.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        items_to_copy = ['iterations',
+                         'pool_profiles']
+        conf = dict((k, cluster[k]) for k in items_to_copy if k in cluster)
+        conf.update(dict(
+            head=localhost,
+            osds=[localhost],
+            osds_per_node=1,
```
Comment (cyx1231st) [inline src/test/crimson/cbt/t2c.py]: `osds_per_node=num_osds,`?

File: src/test/crimson/cbt/radosbench_4K_read.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- desc: |
+   Run radosbench benchmark using cbt.
+   4K read workload.
+
+tasks:
+- cbt:
+    benchmarks:
+      radosbench:
+        concurrent_ops: 4
+        concurrent_procs: 2
```
Comment (cyx1231st) [inline src/test/crimson/cbt/radosbench_4K_read.yaml]: 👍

File: src/script/run-cbt.sh (hunk 1 above)
Comment (tchaikov) [reply src/script/run-cbt.sh]: yeah, i failed to update the driver script. will remove it in the next revision.

File: src/test/crimson/cbt/t2c.py (hunk 2 above)
Comment (tchaikov) [reply src/test/crimson/cbt/t2c.py]: yeah! fixed.

File: src/test/crimson/cbt/t2c.py (hunk 3 above)
Comment (tchaikov) [reply src/test/crimson/cbt/t2c.py]: fixed.

File: src/script/run-cbt.sh [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    local prog_name=$1
+    shift
+    cat <<EOF
+$prog_name [options] <config-file>...
+
+options:
+  -a,--archive-dir    directory in which the test result is stored
+  -h,--help           print this help message
+  --cbt               directory of cbt if you have already a copy of it
+  --use-existing      do not setup/teardown a vstart cluster for testing
```
Comment (rzarzynski) [inline src/script/run-cbt.sh]: :+1: for this feature.

File: src/script/run-cbt.sh [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+usage() {
+    local prog_name=$1
+    shift
+    cat <<EOF
+$prog_name [options] <config-file>...
+
+options:
+  -a,--archive-dir    directory in which the test result is stored
+  -h,--help           print this help message
+  --cbt               directory of cbt if you have already a copy of it
```
Comment (rzarzynski) [inline src/script/run-cbt.sh]: Maybe `--cbt-dir` instead for the sake of being coherent with `--archive-dir`?

File: src/script/run-cbt.sh [hunk 7]
```diff
@@ -0,0 +1,102 @@
+#!/bin/sh
+
+usage() {
```
Comment (rzarzynski) [inline src/script/run-cbt.sh]: nit: the help misses `--build-dir` and `--source-dir`.

File: src/script/run-cbt.sh (hunk 6 above)
Comment (tchaikov) [reply src/script/run-cbt.sh]: yeah, i thought about this. and after a second thought just now, i still think that `--cbt` is better. the reason why i have `archive-dir` or `build-dir` is that i try to be consistent with the variable names defined by autoconf and CMake. but `cbt` here is a specific tool. i intent to differentiate it from the directories we store artifacts or test results.

File: src/script/run-cbt.sh (hunk 7 above)
Comment (tchaikov) [reply src/script/run-cbt.sh]: added.

File: src/test/crimson/cbt/radosbench_4K_read.yaml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+meta:
+- desc: |
+   Run radosbench benchmark using cbt.
+   4K read workload.
+
+tasks:
+- cbt:
+    benchmarks:
+      radosbench:
+        concurrent_ops: 4
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/test/crimson/cbt/radosbench_4K_read.yaml]: changed it to 16.

File: src/test/crimson/cbt/t2c.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        monitoring_profiles = config.get('monitoring_profiles', {})
+        return dict(cluster=self._create_cluster_config(cluster),
+                    benchmarks=benchmarks,
+                    monitoring_profiles=monitoring_profiles)
+
+    def _create_cluster_config(self, cluster):
+        # prepare the "cluster" section consumed by CBT
+        localhost = socket.getfqdn()
+        num_osds = cluster.get('osds_per_node', 3)
+        items_to_copy = ['iterations',
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/test/crimson/cbt/t2c.py]: moved the next line to the end of this line

File: src/test/crimson/cbt/t2c.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/python
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/test/crimson/cbt/t2c.py]: i am not sure about this. but guess it'd more friendly to venv users. changed anyway.

**Comment by tchaikov** [comment]
@rzarzynski updated and repushed.

**Pull request merged**